hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/rmapp/RMAppImpl.java

```diff
@@ -1088,8 +1088,13 @@ public void transition(RMAppImpl app, RMAppEvent event) {
       // otherwise, add it to ranNodes for further process
       app.ranNodes.add(nodeAddedEvent.getNodeId());
 
-      app.logAggregation.addReportIfNecessary(
-          nodeAddedEvent.getNodeId(), app.getApplicationId());
+      if (!nodeAddedEvent.isInAcquiredState()) {
+        app.logAggregation.addReportIfNecessary(
+            nodeAddedEvent.getNodeId(), app.getApplicationId());
+      } else {
+        LOG.warn(String.format("Log aggregation will be skipped for nodeId: %s and appId: %s",
```

Comment:
Please amend the log message to "Not considering container for log aggregation while it is in an ACQUIRED state for nodeId: %s and appId: %s".

Reply:
Thanks for the hint, i fixed it